source/YTEPs/YTEP-0036.rst

```diff
@@ -0,0 +1,248 @@
+
+The second step is to organize the answer tests into classes (e.g., all of the enzo answer tests go in a class called ``TestEnzo``). This not only provides a natural organizational structure within the code itself, but it also provides a simple way of ensuring that all related answer tests get saved to the same answer file. The class structure also facilitates easier running of specific tests, especially in the case where answer tests and unit tests share the same module.
+
+As an extension of the above, each answer test (e.g., ``FieldValuesTest``) that was previously a class of its own is now a method (e.g., ``field_values_test``) of the base ``AnswerTest`` class, which resides in ``yt/utilities/answer_testing/framework.py`` and takes the place of the ``AnswerTestingTest`` class. The use of this base class provides a central location for any new answer tests that might be added in the future. Every class that performs answer testing (e.g. ``TestEnzo``) inherits from ``AnswerTest``, thereby giving it access to the methods of ``AnswerTest``.
+
+This structure removes the need for each individual answer test to have it's own ``__init__`` and ``compare`` methods. The test code is now exclusively what previously resided in the ``run`` method.
+
+Several other changes have been implemented, as well. Previously, when running answer tests that used large data files, the ``--answer-big-data`` option had to be passed to nose. This remains the case, as described below. However, that option was previously passed to the decorators ``requires_ds`` and ``requires_sim``. This is no longer the case. Those decorators now only check to make sure the requested resource exists. The big data files are now handled by a pytest mark ``pytest.mark.big_data``. Further, these decorators used to ``return lambda: None`` whenever the desired resource was not found. Pytest did not recognize these lambdas as valid functions, and so they have been replaced with a ``skip`` method that also employs ``assert False`` in order to mark the test as failing.
+
+The last change made to the code base in switching from nose to pytest is the addition of ``conftest.py`` files.  These are configuration files that are used by pytest in order to define custom fixtures for processes such as setup, teardown, parametrizing tests, and using temporary directories and files.
+
+The primary ``conftest.py`` file resides in the root of the yt repo and it defines the command-line parser as well as the fixtures used across each of the answer tests. Additionally, other ``conftest.py`` files have been defined in each of the frontend directories in order to define fixtures for loading specific data files, and using ``pytest_generate_tests`` to parametrize those answer tests that require it.
+
```

Comment:
I would like to propose to add a paragraph here on means to "test the tests", tools that we can use along the way to help validate the reimplementation of certain tests.

I think this plugin is an example of such tools : it shuffles the execution order of tests each time pytest is called, making sure that we don't accidentally and unknowingly end up writing tests that only happen to pass because of the specific order they are run in, which can happen when tests have global side effects (which can be unpredictable, hard to trace and not desired)
https://pypi.org/project/pytest-randomly/3.4.0/

I'm also thinking we could use coverage reports and badges to get a sense of progression going (even though I know we won't be converging towards 100%, it would still be very satisfying to see the pytest-coverage ratio being bumped with each PR, and let's not undermine the endorphin kick !)

(maybe this belongs to a different section, you decide)

Reply:
I added a section discussing this. Let me know your thoughts!